### DISCUSSION TITLE
code block determines indentation, rathen than indentation determines code block

### DISCUSSION THREAD

**Discussion by NoahStoryM** [racket/rhombus#182] [General]
When I try to program in Python, one of the annoyances is that it uses indentation to indicate a block of code. Indentation represents code blocks sounds very good, it requires programmers to think about the appearance of the code strictly. But in fact, almost all editors support one-click indentation. Generally speaking, programmers don't need to waste too much time on code indentation. But whenever I try to move a piece of code to other places, I have to manually adjust the indentation (I believe this is the editor's job, not the programmers'. In my experience, several problems are caused by accidentally pressing the tab, which is hard to see at the time). What's worse, some codes may mix spaces with \tab to represent indentation.

I think it should be the code block to determine the indentation (so that the editor can help a lot), instead of the indentation to determine the code block (I have encountered some problems caused by this reason, and debugging them later is a very bad experience). 

Python's `:` does look much better than `{`, `begin`, etc.. In C language, `;` is used to indicate the end of a statement. Will it be more elegant to use `:` and `;` to indicate a code block?

**Comment by mflatt** [comment]
For what it's worth, I started with the same opinion. But after trying to implement things and write examples, I ended up in favor of indentation-sensitive parsing (to my surprise).

**Reply by NoahStoryM** [reply]
Indentation sensitivity has brought me some troubles, and to be honest, I haven't experienced its benefits (compared to languages ​​that are not sensitive and can be formatted with one click).  I am a little curious why did you finally decide to adopt it?

**Reply by mflatt** [reply]
My specific experience was writing #119 and then #122. The exercise behind #119 was trying anything that isn't C-style. Going fully to whitespace sensitivity in #122 solved problems that I had in making #119 work well, both in defining something that seemed consistent and in writing examples that looked nice.

My experience goes only so far. I have not yet moved around as much Shrubbry code as S-expression code, and I expect to need a little more tool support than I currently have. So, we'll see.
